test/cloudnative/agentversion_set.go

```diff
@@ -0,0 +1,191 @@
+	builder.Assess("check init containers", checkInitContainers)
+	builder.Assess("check env vars of init container", checkVersionInSampleApp(version))
+}
+
+func assignVersions(t *testing.T, versions []string,
+	old version.SemanticVersion, new version.SemanticVersion) (version.SemanticVersion, version.SemanticVersion) {
+
+	lowerHalf := versions[0 : len(versions)/2]
+	upperHalf := versions[len(versions)/2+1 : len(versions)-1]
+
+	lowerValue := lowerHalf[rand.Intn(len(lowerHalf))]
+	upperValue := upperHalf[rand.Intn(len(upperHalf))]
+
+	lowerVersion, err := version.ExtractSemanticVersion(lowerValue)
+	require.NoError(t, err)
+	newerVersion, err := version.ExtractSemanticVersion(upperValue)
+	require.NoError(t, err)
+
+	compare := version.CompareSemanticVersions(lowerVersion, newerVersion)
+	if compare > 0 {
+		old = newerVersion
+		new = lowerVersion
+	} else if compare < 0 {
+		old = lowerVersion
+		new = newerVersion
+	}
+	return old, new
+}
+
+func updateDynakube(semanticVersion version.SemanticVersion) features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		resources := environmentConfig.Client().Resources()
+		var dk dynatracev1beta1.DynaKube
+
+		err := dynatracev1beta1.AddToScheme(resources.GetScheme())
+		require.NoError(t, err)
+
+		err = resources.Get(ctx, "dynakube", "dynatrace", &dk)
+		require.NoError(t, err)
+
+		dk.Status.UpdatedTimestamp = metav1.Now()
+		dk.Spec.OneAgent.CloudNativeFullStack.Version = semanticVersion.String()
+		err = resources.Update(ctx, &dk)
+		require.NoError(t, err)
+
+		return ctx
+	}
+}
+
+func restartCSIDriver(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+	resources := environmentConfig.Client().Resources()
+	var csiDaemonSet appsv1.DaemonSet
+
+	err := csi.ForEachPod(ctx, resources, func(podItem corev1.Pod) {
+		require.NoError(t, resources.Delete(ctx, &podItem))
+	})
+
+	require.NoError(t, resources.Get(ctx, "dynatrace-oneagent-csi-driver", "dynatrace", &csiDaemonSet))
+	require.NoError(t, wait.For(conditions.New(resources).ResourceMatch(&csiDaemonSet, func(object k8s.Object) bool {
+		daemonSet, isDaemonSet := object.(*appsv1.DaemonSet)
+		return isDaemonSet && daemonSet.Status.NumberReady == daemonSet.Status.NumberAvailable
+	}), wait.WithTimeout(10*time.Minute)))
+
+	require.NoError(t, err)
+	time.Sleep(time.Minute)
+
+	return ctx
+}
+
+func checkVersionInSampleApp(semanticVersion version.SemanticVersion) features.Func {
+	return func(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+		time.Sleep(time.Minute)
+		resources := environmentConfig.Client().Resources()
+		pods := sampleapps.Get(t, ctx, resources)
+
+		for _, podItem := range pods.Items {
+			require.NotNil(t, podItem)
+			require.NotNil(t, podItem.Spec)
+			require.NotNil(t, podItem.Spec.InitContainers)
+
+			for _, container := range podItem.Spec.InitContainers {
```

Comment:
There is at most one `"install-oneagent"`, so instead of looping, it should be found first (using e.g. `slices.Index`) and than processed.